File: backend/app/schemas/notification.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class NotificationItem(BaseModel):
    id: UUID
    type: str
    title: str
    body: str | None = None
    category: str | None = None
    link: str | None = None
    image_url: str | None = None
    action_url: str | None = None
    action_label: str | None = None
    is_read: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class InboxResponse(BaseModel):
    items: list[NotificationItem]
    unread_count: int


class MarkReadResponse(BaseModel):
    updated: int


class PushTokenRequest(BaseModel):
    push_token: str | None = None


class NotificationSettingsResponse(BaseModel):
    id: UUID
    user_id: UUID
    notify_days_before: int
    email_notifications: bool
    push_notifications: bool
    budget_monthly: int | None = None

    model_config = {"from_attributes": True}


class NotificationSettingsUpdateRequest(BaseModel):
    notify_days_before: int | None = Field(default=None, ge=1, le=30)
    email_notifications: bool | None = None
    push_notifications: bool | None = None
    budget_monthly: int | None = None

File: frontend/src/types/notification.ts
export interface NotificationSettings {
  id: string;
  user_id: string;
  notify_days_before: number;
  email_notifications: boolean;
  push_notifications: boolean;
  budget_monthly: number | null;
  /** 푸시를 받을 기기가 연결돼 있는지 (앱에 로그인하면 연결된다) */
  push_device_connected?: boolean;
}

export interface NotificationSettingsUpdate {
  notify_days_before?: number;
  email_notifications?: boolean;
  push_notifications?: boolean;
  budget_monthly?: number | null;
}

export interface NotificationItem {
  id: string;
  type: string;
  title: string;
  body: string | null;
  category: string | null;
  link: string | null;
  image_url: string | null;
  action_url: string | null;
  action_label: string | null;
  is_read: boolean;
  created_at: string;
}

export interface InboxResponse {
  items: NotificationItem[];
  unread_count: number;
}
